core/src/trezor/ui/layouts/tt/__init__.py

```diff
@@ -518,6 +520,31 @@ async def confirm_output(
     await raise_if_cancelled(interact(ctx, content, "confirm_output", br_code))
 
 
+async def confirm_payment_request(
+    ctx: wire.GenericContext,
+    recipient_name: str,
+    amount: str,
+    memos: list[str],
+) -> bool:
+    text = Text("Confirm sending", ui.ICON_SEND, ui.GREEN)
+    text.normal(amount + " to")
+    text.normal(recipient_name)
+    text.br_half()
+    for memo in memos:
+        text.normal(memo)
+
+    result = await raise_if_cancelled(
+        interact(
+            ctx,
+            InfoConfirm(text, info="Details"),
```

Comment:
what is supposed to happen if you click Details?

also given that there can be multiple memos, the screen should probably be paginated right from the start

Reply:
> what is supposed to happen if you click Details?

It results in `show_payment_req_details` being set to `true`, which means each output in the payment request is confirmed individually, see https://github.com/trezor/trezor-firmware/pull/2069/files#diff-5d7c5fcd1f383543bfa486901d66e95191cb54deb21fbd3449a4489dffddf9c9R190-R193.

> also given that there can be multiple memos, the screen should probably be paginated right from the start

You mean using `paginate_paragraphs()`?

Reply:
Oh, I understand now. This is confusing because the function is called `confirm_...` so one would assume that the boolean result is confirmed yes/no.

Please either change the function to return `result` (and then `show_payment_req_details = result is INFO`), or rename to `should_show_details` or something, like `should_show_more` below.

> You mean using `paginate_paragraphs()`?

yes

Reply:
> Please either change the function to return `result` [...]

https://github.com/trezor/trezor-firmware/pull/2069/commits/741087054d607bec797ce465dab1716e2d0e783a

> also given that there can be multiple memos, the screen should probably be paginated right from the start

https://github.com/trezor/trezor-firmware/pull/2069/commits/21962b8fae72ac58017c14cb2d82c828006be4ea